### PR TITLE
RD-2898 Loosen assertions in topology spec

### PR DESCRIPTION
**Pull request by Gelio**
Refactor the assertions in `topology_spec` to allow extra properties in the asserted objects. Changes the existing logic of using `omit` to skip some fields and now asserts that only the fields from the expected objects match.

## System tests

https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1002/pipeline

Queued 2021-07-29 9:56 CEST

### CODE REVIEW

File: test/cypress/integration/widgets/topology_spec.ts [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+/**
+ * Performs a deep equality check assuming the `assertedValue` can
+ * contain extra properties not present in `expectedValue`
+ */
+// NOTE: TS does not allow using `Record<string, object>` for the `assertedValue`
+// eslint-disable-next-line @typescript-eslint/ban-types
+const expectToContainSubset = (assertedValue: object, expectedValue: Record<string, unknown>) => {
+    const partialValueToCompare = pick(assertedValue, Object.keys(expectedValue));
+    expect(partialValueToCompare).to.deep.equal(expectedValue);
+};
```
Comment (Gelio) [inline test/cypress/integration/widgets/topology_spec.ts:13]: There is https://www.npmjs.com/package/chai-subset, but I decided that writing our own utility function here would be easier and less complex

**Review by qooban — APPROVED** [review]
Seems fine. I assume `master` branch will get this change, too.

**Comment by Gelio** [comment]
> Seems fine. I assume `master` branch will get this change, too.

Yup, I'll cherry-pick after merging this PR :slightly_smiling_face:

**Comment by Gelio** [comment]
System tests passed, the PR is relatively small, so I'll merge it with just a single approval

**Pull request merged**